pkg/config/config_parser.go

```diff
@@ -236,3 +241,72 @@ func decodeCloudQueryBlock(block *hcl.Block, ctx *hcl.EvalContext) (CloudQuery,
 	cq.Logger = &logging.GlobalConfig
 	return cq, gohcl.DecodeBody(block.Body, ctx, &cq)
 }
+
+func validateConnection(connection *Connection) diag.Diagnostics {
+	var diags diag.Diagnostics
+	if connection.DSN != "" {
+		// allow using a DSN flag even if the config file has explicitly attributes (user, password, host, database, etc.)
+		dsnFromFlag := viper.GetString("dsn")
+		if dsnFromFlag == "" && connection.IsAnyConnParamsSet() {
+			diags = append(diags, diag.NewBaseError(
+				fmt.Errorf("invalid connection configuration"),
+				diag.USER, diag.WithOptionalSeverity(diag.ERROR),
+				diag.WithDetails("DSN specified along with explicit attributes, only one type is supported")),
+			)
+		}
+	} else {
```

Comment:
this function can be simplified in the first if, its enough to just return if the first if occurs otherwise return nil, and then you don't need the else.

Reply:
Great catch! I would expect the linter to report it actually, not sure why it didn't.
Fixed in 6d4caed5dec28aeb754b12ed0e964aa4d1e92528